### ISSUE TITLE
235206706 Don't add ? to blank person_sex field in baptism records during processing

### ISSUE THREAD

**Issue by ghost** [FreeUKGen/FreeUKRegProductIssues#1480]
Issue reported by **FR Somerset Coordinator** at 2022-06-15 07:11:46 UTC
Time: 2022-06-15T07:06:22+00:00
Session ID: 
Problem Page URL: []()
Previous Page URL: [https://www.freereg.org.uk/contacts/new](https://www.freereg.org.uk/contacts/new)
Reported Issue:
Please see my reply of yesterday to contact no 235083426 on the contacts system.

This is one of the many many cases where a question mark is recorded in the FreeREG record instead of the sex of the child baptised.

As far as I can see an explanation of the reason for this particular kind of ? is not given anywhere in the Researcher Help pages, only the explanation for a question mark eg at the end of a name or in UCF.

So, perhaps it should be added to the Help pages?

Although, as I said to the contact, despite there being millions of such ?s in the FR records, this is the first time someone has written in to me about one!

Jackie James

**Comment by PatReynolds** [comment]
Moving forwards people should leave blank. 

Need to remove "?" leaving blank, M and F as three remaining options.

@SteveBiggs  notes Winreg inserts a dash. 
@PatReynolds to add to Unitt definition (blank)

**Comment by SteveBiggs** [comment]
In WinREG, for the sex field, you can select either Male, Female, Unknown or blank. If you select Unknown, WinREG puts a hyphen "-" in the field and when the file is uploaded to FreeREG, it remains as a hyphen. FreeREG accepts this without reporting an error, however when you edit a record online in FreeREG you are presented only with either M, F, ? or blank.

The Help section on FreeREG states you should enter:

- a single character from f, F, m, or M
- leave blank where the sex is not stated
- if the entry is illegible, put ?
- if you find just a ‘-’ (a hyphen) or a ‘?’ in a transcript, put ?

**Comment by DeniseColbert** [comment]
Once transcribers are no longer using WinREG, a rake task to find all "-" in sex field and replaced with "?"

**Comment by stsccfr** [comment]
I don't think we should use ? in any situation where the document itself has nothing (i.e., is blank).  A ? mark could just as easily mean that there is something written there but the transcriber can't read it, but that's different than nothing being there.  Our use of ? at the end of words to indicate uncertainty makes sense, and I think we should retain that particular usage (it's already documented in the UCF Help page).  I would like to alter the processing code so that we do not add a ? mark to the Sex field if it is blank.  If we then decide to clean up the existing CSV files, I would like to see us remove the ? mark from the Sex field.

**Comment by stsccfr** [comment]
I think the offending code is in new_freereg_csv_update_processor.rb in the function process_baptism_sex_field() where we do this:

```
  def  process_baptism_sex_field(field)
    case
    when field.nil?
      return_field = "?"
```

**Comment by DeniseColbert** [comment]
Ready to be tested on Test3 @stsccfr

**Comment by stsccfr** [comment]
I uploaded a WinREG file and a FlexCSV file to test3 containing records with and without any Sex value.  The processor changes all valid Sex values to either 'M' or 'F', and leaves the Sex field blank if there is no value in it.  Ready for Deployment.

Incidentally, what is considered a valid Sex value is defined in `freereg_validations.rb` as follows:

`VALID_MALE_SEX = ["M","M." ,"SON","MALE","MM","SON OF"]`
`VALID_FEMALE_SEX = ["F","FF","FFF","FM","F.","FEMALE","DAUGHTER","WIFE","DAUGHTER OF","DAU", "DAU OF"]`

Note that the above definitions are used independent of case, so 'm' and 'male' are also valid, etc.

**Comment by DeniseColbert** [comment]
Done: closing
